tests/cross-compile/loop-attribs.slang

```diff
@@ -0,0 +1,19 @@
+// loop-attribs.slang
+// Test that loop attributes are correctly emitted to the resulting HLSL.
+
+//TEST:SIMPLE:-target hlsl  -entry main -stage fragment -profile sm_6_0 
```

Comment:
It is usually best to avoid having tests where the expected output is the generated HLSL or GLSL text from the compiler, because that can end up changing in a bunch of ways that shouldn't affect the validity of a given test.

My preference for stuff like this is to define is as a `//TEST:CROSS_COMPILE:` test and then have a `*.slang.hlsl` file that contains the equivalent raw HLSL (you could use your current `.slang.expected` to set that up), and then the test will check against the expected output DXBC and/or DXIL.

Under that model, we don't care if the HLSL produced by Slang is exactly the same; we just care if the output HLSL compiles to the same DXBC/DXIL.

(Note that I'm also *not* advocating for establishing a raw DXBC/DXIL baseline, since that is also a source of problems since fxc/dxc version changes can alter the exact DXBC/DXIL produced. It might help to hash out the details here in a separate conversation thread...)